java-generator/core/src/main/java/io/fabric8/java/generator/WritableCRCompilationUnit.java

```diff
@@ -28,44 +27,35 @@
 
 public class WritableCRCompilationUnit {
 
-    private final CompilationUnit cu;
-    private final List<String> classNames;
+    private final List<GeneratorResult.ClassResult> classResults;
 
-    WritableCRCompilationUnit(CompilationUnit cu, List<String> classNames) {
-        this.cu = cu;
-        this.classNames = classNames;
+    public List<GeneratorResult.ClassResult> getClassResults() {
+        return classResults;
     }
 
-    public void writeAllJavaClasses(File basePath) {
+    WritableCRCompilationUnit(List<GeneratorResult.ClassResult> classResults) {
+        this.classResults = classResults;
+    }
+
+    public void writeAllJavaClasses(File basePath, Optional<String> basePackage) {
```

Comment:
Just sayin' here... 
According to a [comment](https://github.com/fabric8io/kubernetes-client/pull/3801#discussion_r801389053) in a recent PR review, using `Optional`s as method parameters is not recommended.
BTW I can spot one between `createFolders()` arguments, too.